kyuubi-main/src/main/scala/org/apache/kyuubi/operation/KyuubiOperationManager.scala

```diff
@@ -49,6 +50,17 @@ class KyuubiOperationManager private (name: String) extends OperationManager(nam
     tSessionHandle
   }
 
+  private def getQueryTimeout(statementTimeout: Long): Long = {
+    // If a timeout value `queryTimeout` is specified by users and it is smaller than
+    // a session timeout value, we use the user-specified value.
+    val sessionTimeout = getConf.get(KyuubiConf.OPERATION_QUERY_TIMEOUT)
+    if (sessionTimeout > 0 && (statementTimeout <= 0 || sessionTimeout < statementTimeout)) {
```

Comment:
what if sessionTimeout=-1 and statementTimeout=-1?

Reply:
nothing happened, that only add timeout checkout when timeout > 0.